modules/indexer/code/bleve/bleve.go

```diff
@@ -217,19 +216,14 @@ func (b *Indexer) addDelete(filename string, repo *repo_model.Repository, batch
 func (b *Indexer) Index(ctx context.Context, repo *repo_model.Repository, sha string, changes *internal.RepoChanges) error {
 	batch := inner_bleve.NewFlushingBatch(b.inner.Indexer, maxBatchSize)
 	if len(changes.Updates) > 0 {
-		r, err := gitrepo.OpenRepository(ctx, repo)
+		gitBatch, err := git.NewBatch(ctx, repo.RepoPath())
 		if err != nil {
 			return err
 		}
-		defer r.Close()
-		gitBatch, err := r.NewBatch(ctx)
-		if err != nil {
-			return err
-		}
-		defer gitBatch.Close()
```

Comment:
I do not think this change is right.

In most cases, you still need to use `defer` in case panic occurs.

Reply:
7ac68add39